pandas/core/arrays/arrow/dtype.py

```diff
@@ -90,7 +100,36 @@ def type(self):
         """
         Returns pyarrow.DataType.
         """
-        return type(self.pyarrow_dtype)
+        pa_type = self.pyarrow_dtype
+        if pa.types.is_integer(pa_type):
```

Comment:
Could you add decimal while you're changing this?

Reply:
updated+green